src/qgis_geonode/gui/geonode_source_select_provider.py

```diff
@@ -542,4 +558,106 @@ def update_keywords(self, keywords: typing.Optional[typing.List[str]] = None):
         if keywords:
             self.keyword_cmb.addItem("")
             self.keyword_cmb.addItems(keywords)
+            self.save_settings()
         self.message_bar.clearWidgets()
+
+    def restore_settings(self):
```

Comment:
As stated above, please use a more helpful name for this method

```suggestion
    def restore_search_filters(self):
```